pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -159,22 +152,21 @@ func getMonitoringConsoleDeployment(cr splcommon.MetaObject, spec *enterprisev1.
 		},
 	}
 
-	env := []corev1.EnvVar{
-		{Name: "SPLUNK_CONFIGMAP_HASH",
-			Value: configMapHash,
-		},
-	}
+	env := []corev1.EnvVar{}
+
 	// append labels and annotations from parent
-	splcommon.AppendParentMeta(deploymentMC.Spec.Template.GetObjectMeta(), cr.GetObjectMeta())
+	splcommon.AppendParentMeta(statefulSet.Spec.Template.GetObjectMeta(), cr.GetObjectMeta())
 
 	// update statefulset's pod template with common splunk pod config
-	updateSplunkPodTemplateWithConfig(&deploymentMC.Spec.Template, cr, spec, instanceType, env, secretName)
+	updateSplunkPodTemplateWithConfig(client, &statefulSet.Spec.Template, cr, spec, instanceType, env, secretName)
+
+	statefulSet.SetOwnerReferences(append(statefulSet.GetOwnerReferences(), splcommon.AsOwner(cr, true)))
 
-	return deploymentMC, nil
+	return statefulSet, nil
 }
 
-//ApplyMonitoringConsoleEnvConfigMap creates or updates a Kubernetes ConfigMap for extra env for monitoring console pod
-func ApplyMonitoringConsoleEnvConfigMap(client splcommon.ControllerClient, namespace string, crName string, newURLs []corev1.EnvVar, addNewURLs bool) (*corev1.ConfigMap, error) {
+//getMonitoringConsoleEnvConfigMap creates or updates a Kubernetes ConfigMap for extra env for monitoring console pod
```

Comment:
Nit: any reason we changed it to get vs Apply. Apply seems more appropriate since we are create/updating here.

Reply:
getMonitoringConsoleEnvConfigMap looks more appropriate since we are not "applying" anything here, we just create/update and get the update configMap back

Reply:
Apply implies create/update and we are following the same everywhere I guess. Eg. ApplyNamespaceScopedSecret, ApplyConfigMap, ApplySecret etc.. I think the paradigm followed in these APIs is:

```
err := Get()
if err == nil {
     if change {
          update
     }
} else {
     create
}
```

Minor observation to follow the existing pattern. :)

Reply:
Ok, @akondur reverted it back.